app/controllers/ReportingController.scala

```diff
@@ -46,24 +46,34 @@ trait ReportingController extends BaseController {
 
   val appRepository: GeneralApplicationRepository
   val cdRepository: ContactDetailsRepository
-  val reportingRepository: ReportingRepository
   val authProviderClient: AuthProviderClient
   val questionnaireRepository: QuestionnaireRepository
   val testReportRepository: TestReportRepository
   val assessmentScoresRepository: ApplicationAssessmentScoresRepository
+  val medRepository: MediaRepository
 
   def candidateProgressReport(frameworkId: String) = Action.async { implicit request =>
     appRepository.candidateProgressReport(frameworkId).map(r => Ok(Json.toJson(r)))
   }
 
-  def retrieveDiversityReport = Action.async { implicit request =>
-    reportingRepository.findLatest().map {
-      case Some(report) => Ok(Json.toJson(report))
-      case None => NotFound
+  def diversityReport(frameworkId: String) = Action.async { implicit request =>
+    val reports = for {
+      applications <- appRepository.diversityReport(frameworkId)
+      questionnaires <- questionnaireRepository.findAllForDiversityReport
+      medias <- medRepository.findAll()
+    } yield {
+      for {
+        a <- applications
+        q <- Some(questionnaires.get(a.applicationId))
```

Comment:
questionnnaires.get(a.applicationId) already returns an option. Why is this line not:

q = questionnaires.get(a.applicationId) ?